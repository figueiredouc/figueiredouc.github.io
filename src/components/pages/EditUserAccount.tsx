import 'app/App.css';

import React, { useState } from 'react';
import {
  createStyles,
  makeStyles,
  Theme,
  Typography,
  Container,
  Divider,
  Button,
} from '@material-ui/core';
import FormControl from '@material-ui/core/FormControl';
import IconButton from '@material-ui/core/IconButton';
import Input from '@material-ui/core/Input';
import InputAdornment from '@material-ui/core/InputAdornment';
import InputLabel from '@material-ui/core/InputLabel';
import { fade } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Visibility from '@material-ui/icons/Visibility';
import VisibilityOff from '@material-ui/icons/VisibilityOff';
import AppPage from 'components/shared/AppPage';
import loggedUser from '__mocks__/loggedUser.json';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      height: 'inherit',
      flexGrow: 1,
      paddingTop: theme.spacing(8),
      paddingLeft: 0,
      paddingRight: 0,
      '& .MuiButton-root': {
        backgroundColor: 'transparent',
      },
      '& .MuiButton-root:hover': {
        backgroundColor: fade('#660000', 0.3),
      },
      '& .MuiTypography-body1': {
        paddingTop: theme.spacing(2),
      },
    },
    name: {
      color: '#033860',
    },
    divider: {
      backgroundColor: '#B80C09',
    },
    infoUser: {
      display: 'flex',
      justifyContent: 'space-between',
    },
    sideText: {
      marginLeft: theme.spacing(2),
    },
    saveButton: {
      marginTop: theme.spacing(2),
      backgroundColor: fade('#660000', 0.2),
      '&:hover': {
        backgroundColor: fade('#660000', 0.3),
      },
    },
    rootForm: {
      '& .MuiTextField-root': {
        marginTop: theme.spacing(1),
      },
      '& .MuiFormControlLabel-root': {
        marginTop: theme.spacing(2),
      },
    },
  })
);

interface Password {
  password: string;
  confirmPassword: string;
  showPassword: boolean;
  showConfirmPassword: boolean;
}

type ChangeEvent = React.ChangeEvent<HTMLInputElement>;

const EditUserAccount = () => {
  const classes = useStyles();

  const [userDetails, setUserDetails] = useState(loggedUser);

  const [passwordValues, setPasswordValues] = useState<Password>({
    password: '',
    confirmPassword: '',
    showPassword: false,
    showConfirmPassword: false,
  });

  const handlePasswordChange = (prop: keyof Password) => (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    setPasswordValues({ ...passwordValues, [prop]: event.target.value });
  };

  const handleClickShowPassword = () => {
    setPasswordValues({
      ...passwordValues,
      showPassword: !passwordValues.showPassword,
    });
  };

  const handleMouseDownPassword = (
    event: React.MouseEvent<HTMLButtonElement>
  ) => {
    event.preventDefault();
  };

  const handleClickShowConfirmPassword = () => {
    setPasswordValues({
      ...passwordValues,
      showConfirmPassword: !passwordValues.showConfirmPassword,
    });
  };

  const handleMouseDownConfirmPassword = (
    event: React.MouseEvent<HTMLButtonElement>
  ) => {
    event.preventDefault();
  };

  const handleChange = (name: string, e: ChangeEvent) => {
    setUserDetails({ ...userDetails, [name]: e.target.value });
  };

  return (
    <AppPage>
      <Container className={classes.root}>
        <div className={classes.infoUser}>
          <Typography className={classes.name} variant="h5">
            {loggedUser.name}
          </Typography>
        </div>
        <Divider className={classes.divider} />
        <FormControl className={classes.rootForm}>
          <TextField
            fullWidth
            id="name"
            label="Nome/Empresa"
            value={userDetails.name}
            onChange={(e: ChangeEvent) => handleChange('name', e)}
          />
          <TextField
            fullWidth
            id="email"
            label="Email"
            value={userDetails.email}
            onChange={(e: ChangeEvent) => handleChange('email', e)}
          />
          <div>
            <TextField
              id="nif"
              label="NIF"
              value={userDetails.nif}
              onChange={(e: ChangeEvent) => handleChange('nif', e)}
            />
            <TextField
              className={classes.sideText}
              id="telefone"
              label="Telefone"
              value={userDetails.telefone}
              onChange={(e: ChangeEvent) => handleChange('telefone', e)}
            />
          </div>
          <TextField
            fullWidth
            id="morada"
            label="Morada"
            value={userDetails.morada}
            onChange={(e: ChangeEvent) => handleChange('morada', e)}
          />
          <FormControl className={classes.rootForm}>
            <InputLabel htmlFor="standard-adornment-password">
              Password
            </InputLabel>
            <Input
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                    aria-label="toggle password visibility"
                    onClick={handleClickShowPassword}
                    onMouseDown={handleMouseDownPassword}
                  >
                    {passwordValues.showPassword ? (
                      <Visibility />
                    ) : (
                      <VisibilityOff />
                    )}
                  </IconButton>
                </InputAdornment>
              }
              id="standard-adornment-password"
              type={passwordValues.showPassword ? 'text' : 'password'}
              value={passwordValues.password}
              onChange={handlePasswordChange('password')}
            />
          </FormControl>
          <FormControl className={classes.rootForm}>
            <InputLabel htmlFor="standard-adornment-confirm-password">
              Confirmar Password
            </InputLabel>
            <Input
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                    aria-label="toggle password visibility"
                    onClick={handleClickShowConfirmPassword}
                    onMouseDown={handleMouseDownConfirmPassword}
                  >
                    {passwordValues.showConfirmPassword ? (
                      <Visibility />
                    ) : (
                      <VisibilityOff />
                    )}
                  </IconButton>
                </InputAdornment>
              }
              id="standard-adornment-confirme-password"
              type={passwordValues.showConfirmPassword ? 'text' : 'password'}
              value={passwordValues.confirmPassword}
              onChange={handlePasswordChange('confirmPassword')}
            />
          </FormControl>
          <Button
            className={classes.saveButton}
            variant="outlined"
            onClick={() => {
              /* TODO */
            }}
          >
            Guardar
          </Button>
        </FormControl>
      </Container>
    </AppPage>
  );
};

export default EditUserAccount;
