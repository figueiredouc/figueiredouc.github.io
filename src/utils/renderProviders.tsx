/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom'; // 4.0.0
import { render } from '@testing-library/react';
import { AuthContext, AuthStore } from 'context/AuthContext';
import {
  BreadcrumbsContext,
  BreadcrumbsStore,
} from 'context/BreadcrumbsContext';
import { ShopCartContext, ShopCartStore } from 'context/ShopCartContext';
import {
  linkedBreadcrumbs as defaultLinkedB,
  activeBreadcrumb as defaultActiveB,
} from '__mocks__/breadcrumbs';
import { shopCart as defaultShopCart } from '__mocks__/products';

export const defaultStore = {
  AuthStore,
  ShopCartStore,
  BreadcrumbsStore,
};

const customRender = (ui: any, { store = defaultStore, ...options } = {}) => {
  const CombinedProviders = ({ children }: any) => {
    return (
      <AuthContext.Provider value={store.AuthStore()}>
        <ShopCartContext.Provider value={store.ShopCartStore()}>
          <BreadcrumbsContext.Provider value={store.BreadcrumbsStore()}>
            <MemoryRouter>{children}</MemoryRouter>
          </BreadcrumbsContext.Provider>
        </ShopCartContext.Provider>
      </AuthContext.Provider>
    );
  };

  return render(ui, { wrapper: CombinedProviders, ...options });
};

export const customStore = ({
  isLoadingAuthStore = false,
  getLoggedUser = jest.fn(),
  createUser = jest.fn(),
  loggedUser = { user: 'xpto', id: 1, token: '123' },
  shopCart = defaultShopCart,
  shopCartTotalPrices = {
    totalPrice: '20',
    totalPriceBeforeTax: '40',
  },
  linkedBreadcrumbs = defaultLinkedB,
  activeBreadcrumb = defaultActiveB,
  useDefaultAuthStore = false,
  useDefaultShopCartStore = false,
  useDefaultBreadcrumbsStore = false,
} = {}) => {
  const newAuthStore = () => ({
    isLoadingUser: isLoadingAuthStore,
    getLoggedUser,
    loggedUser,
    createUser,
  });

  const newShopCartStore = () => ({
    shopCart,
    shopCartTotalPrices,
  });

  const newBreadcrumbsStore = () => ({
    linkedBreadcrumbs,
    activeBreadcrumb,
  });

  return {
    ...defaultStore,
    AuthStore: useDefaultAuthStore ? defaultStore.AuthStore : newAuthStore,
    ShopCartStore: useDefaultShopCartStore
      ? defaultStore.ShopCartStore
      : newShopCartStore,
    BreadcrumbsStore: useDefaultBreadcrumbsStore
      ? defaultStore.BreadcrumbsStore
      : newBreadcrumbsStore,
  };
};

export * from '@testing-library/react';
export { customRender as render };
