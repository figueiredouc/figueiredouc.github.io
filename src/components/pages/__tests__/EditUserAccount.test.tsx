import React from 'react';
import { render } from 'utils/renderProviders';
import EditUserAccount from 'components/pages/EditUserAccount';

describe('renders EditUserAccount', () => {
  it('should match snapshot', () => {
    const { container } = render(<EditUserAccount />);

    expect(container).toMatchSnapshot();
  });
});
