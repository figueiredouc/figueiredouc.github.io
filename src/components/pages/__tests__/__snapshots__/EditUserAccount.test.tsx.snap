// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EditUserAccount should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-11"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-13 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-header-14 MuiToolbar-gutters"
      >
        <div
          class="makeStyles-logoContainer-15"
        >
          <button
            aria-controls="long-menu"
            aria-haspopup="true"
            aria-label="more"
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <img
            alt="cmai"
            height="50px"
            src="logo.png"
          />
        </div>
        <div
          class="makeStyles-flex-21"
        >
          <div
            class="makeStyles-search-23"
          >
            <div
              class="makeStyles-searchIcon-24"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root makeStyles-inputRoot-25"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input makeStyles-inputInput-26"
                placeholder="Search…"
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
              </span>
              Iniciar Sessão
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
  <div
    class="makeStyles-dividerMargin-17"
  >
    <hr
      class="MuiDivider-root makeStyles-divider-16"
    />
  </div>
  <div
    class="MuiContainer-root makeStyles-container-8 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg"
    >
      <div
        class="makeStyles-infoUser-4"
      >
        <h5
          class="MuiTypography-root makeStyles-name-2 MuiTypography-h5"
        >
          Ricardo Araújo Pereira
        </h5>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider-3"
      />
      <div
        class="MuiFormControl-root makeStyles-rootForm-7"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
            for="name"
            id="name-label"
          >
            Nome/Empresa
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="name"
              type="text"
              value="Ricardo Araújo Pereira"
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
            for="email"
            id="email-label"
          >
            Email
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="email"
              type="text"
              value="ricardoapereira@gmail.com"
            />
          </div>
        </div>
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
              for="nif"
              id="nif-label"
            >
              NIF
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                id="nif"
                type="text"
                value="336080116"
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-sideText-5"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
              for="telefone"
              id="telefone-label"
            >
              Telefone
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                id="telefone"
                type="text"
                value="912345678"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
            for="morada"
            id="morada-label"
          >
            Morada
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="morada"
              type="text"
              value="Rua algures em Lisboa, n345, 1D / 5695-233"
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root makeStyles-rootForm-7"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="standard-adornment-password"
          >
            Password
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              id="standard-adornment-password"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root makeStyles-rootForm-7"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="standard-adornment-confirm-password"
          >
            Confirmar Password
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              id="standard-adornment-confirme-password"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-saveButton-6"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Guardar
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
